From 9627e77b733a0585dd49df60b4083b56385e56f6 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Thu, 9 Jan 2025 10:55:51 +0100
Subject: [PATCH 22/55] base-files: add ucidef_set_wireless_macaddr_base()

Adds UCI defaults helper function to set base MAC address for wireless bands, enabling per-band MAC address customisation during board initialisation.

Signed-off-by: John Crispin <john@phrozen.org>
---
 .../files/lib/functions/uci-defaults.sh       | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/package/base-files/files/lib/functions/uci-defaults.sh b/package/base-files/files/lib/functions/uci-defaults.sh
index 3155940e10..c0c4b22262 100644
--- a/package/base-files/files/lib/functions/uci-defaults.sh
+++ b/package/base-files/files/lib/functions/uci-defaults.sh
@@ -366,6 +366,27 @@ ucidef_set_label_macaddr() {
 	json_select ..
 }
 
+ucidef_set_wireless_macaddr_base() {
+	local band="$1"
+	local macaddr_base="$2"
+
+	case "$band" in
+		2g|5g|6g) ;;
+		*) return;;
+	esac
+	[ -z "$macaddr_base" ] && return
+
+	json_select_object wlan
+		json_select_object defaults
+			json_select_object ssids
+				json_select_object "$band"
+					json_add_string macaddr_base "$macaddr_base"
+				json_select ..
+			json_select ..
+		json_select ..
+	json_select ..
+}
+
 ucidef_add_atm_bridge() {
 	local vpi="$1"
 	local vci="$2"
-- 
2.34.1

